File: rpc_basic/tests/sf05.c
int SF05_CheckCrc(u8t data[], u8t nbrOfBytes, u8t checksum){
//==============================================================================
  u8t bit;     // bit mask
  u8t crc = 0; // calculated checksum
  u8t byteCtr; // byte counter
  
  // calculates 8-Bit checksum with given polynomial
  for(byteCtr = 0; byteCtr < nbrOfBytes; byteCtr++)
  {
    crc ^= (data[byteCtr]);
    for(bit = 8; bit > 0; --bit)
    {
      if(crc & 0x80) crc = (crc << 1) ^ POLYNOMIAL;
      else           crc = (crc << 1);
    }
  }
  
  // verify checksum
  if(crc != checksum) return CHECKSUM_ERROR;
  else                return NO_ERROR;
}

File: app_sensors/tests/sf05.nim
proc sF05CheckCrc*(data: ptr U8t; nbrOfBytes: U8t; checksum: U8t): cint =
  ## ==============================================================================
  var bit: U8t
  ##  bit mask
  var crc: U8t = 0
  ##  calculated checksum
  var byteCtr: U8t
  ##  byte counter
  ##  calculates 8-Bit checksum with given polynomial
  byteCtr = 0
  while byteCtr < nbrOfBytes:
    crc = crc xor (data[byteCtr])
    bit = 8
    while bit > 0:
      if crc and 0x80:
        crc = (crc shl 1) xor polynomial
      else:
        crc = (crc shl 1)
      dec(bit)
    inc(byteCtr)
  ##  verify checksum
  if crc != checksum:
    return checksum_Error
  else:
    return no_Error
  